### PR TITLE
NETOBSERV-1425: enhance metrics filters

### PR DESCRIPTION
**Pull request by jotak**
## Description

- New filters: not_equal, not_match_regex
- **Breaking change**: renamed filter types **exact** => **equal**, **regex** => **match_regex**
- Variable interpolation: e.g. "api.MetricsFilter{Key: "src-ns", Value: "$(dst-ns)"}" to filter for intra-namespace traffic

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [x] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -20,19 +20,23 @@ Following is the supported API format for prometheus encode:
                  filter: an optional criterion to filter entries by. Deprecated: use filters instead.
                      key: the key to match and filter by
                      value: the value to match and filter by
-                     type: (enum) the type of filter match: exact (default), presence, absence or regex
-                         exact: match exactly the provided fitler value
+                     type: (enum) the type of filter match: exact (default), exact_not, presence, absence, regex or regex_not
+                         exact: match exactly the provided filter value
+                         exact_not: the value must be different from the provided filter
```
Comment (jpinsonneau) [inline docs/api.md:25]: `different` ?

Comment (jpinsonneau) [reply docs/api.md:25]: or maybe we can rename both to `equal` / `not_equal`

That will allow to extrapolate on any cases such as `equal_ignore_case`...

File: docs/api.md [hunk 2]
```diff
@@ -20,19 +20,23 @@ Following is the supported API format for prometheus encode:
...
                          presence: filter key must be present (filter value is ignored)
                          absence: filter key must be absent (filter value is ignored)
                          regex: match filter value as a regular expression
+                         regex_not: the filter value must not match the provided regular expression
```
Comment (jpinsonneau) [inline docs/api.md:29]: Same here, I would prefer `match_regex` / `not_match_regex`

**Review by jpinsonneau — COMMENTED** [review]
It seems we worked on the same thing in parallel:
https://github.com/netobserv/flowlogs-pipeline/pull/605 :upside_down_face: 

Let's see how we name the filters and I'll close my PR after that

File: docs/api.md (hunk 1 above)
Comment (jotak) [reply docs/api.md:25]: I'm fine with this renaming, `equal`/`not_equal` sounds more familiar

**Review by jpinsonneau — APPROVED** [review]
Looks good to me ! Thanks @jotak

**Comment by jpinsonneau** [comment]
@jotak do you have an Operator PR for these changes ?

**Pull request merged**